### PR TITLE
Pan zoom mnrva 227

### PR DESCRIPTION
**Pull request by pras7430**
## JIRA:
See ticket here - https://jira.rax.io/browse/MNRVA-227

 ## Description:

1. Zoom Feature added to Graph (using metricname , tags, and timestamp).
2. Fixed default graph loading for "npm run dev".
3.  selection of Metricgroup or MetricName.
zoom

 ## Testing:

 ## Screenshots:
(if applicable)

### CODE REVIEW

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts [hunk 1]
```diff
@@ -276,32 +293,61 @@ export class VisualizePage {
...
    * @param qryPrmHndlr Params
    */
-  changingQueryParams(data: Params) {
+  changingQueryParams(data: Params, queryPramsHandler:any) {
     this.router.navigate([],
       {
         relativeTo: this.route,
         queryParams: data,
-        queryParamsHandling: 'merge', // remove to replace all query params by provided
+        queryParamsHandling: queryPramsHandler, // remove to replace all query params by provided
```
Comment (FarajiA) [inline src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts:331]: Ahh okay I had removed this. It didn't look like any other values were needed other than `'merge'`.

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts [hunk 2]
```diff
@@ -276,32 +293,61 @@ export class VisualizePage {
...
+      start : new Date(event.start).toISOString(),
+    end : new Date(event.end).toISOString(),
+     
+      ...(!!this.metricServie.selectedTags && { tag: this.metricServie.selectedTags.tag }),
+      ...(!!event.metricName && { metricName: event.metricName }),
+      ...(!!this.selectedGroup && { metricGroup: undefined})
+    }
+    
+    this.metricServie.getMetricsDataPoints(query).subscribe((data) => {
+      this.graphMetric.filter(a => a.metric ===event.metricName)[0].graph= data;
```
Comment (FarajiA) [inline src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts:315]: You may have to explain this one to me.

File: src/app/_shared/components/hedwig-graph/hedwig-graph.component.html [hunk 3]
```diff
@@ -10,7 +10,9 @@
 
 <div class="hxRow">
   <div class="hxCol hxSpan-12">
-    <hedwig-graph [attr.data-type]="type" [attr.data-group]="group" [attr.data-field]="fieldName"
+    <hedwig-graph [attr.data-type]="type" [attr.data-group]="group" (timeStampSelection)="timeStampReceiver($event)"
```
Comment (FarajiA) [inline src/app/_shared/components/hedwig-graph/hedwig-graph.component.html:13]: Yes, we added `unit`

**Review by FarajiA — APPROVED** [review]
This is coming together nicely

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts (hunk 2 above)
Comment (pras7430) [reply src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts:315]: we have All Data in "This.graphMetrich" with multiple metricName, while doing zoom to particular graph, queryApi is being called with selected(zoomed) graph metricname, selected tags and the selected time span.
Response of queryAPI for selected(zoomed) graph will replace data in  "This.graphMetrich" for the same metric name only.

I am pretty sure some cases query API will respond with multiple data in that case data will be dirty. we asked this question in Engineering sync, so maybe they add any "unique identification" for each Data.. in that case, we need to do rework on this filtration part.

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts (hunk 1 above)
Comment (pras7430) [reply src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts:331]: Yeah, I did notice when you removed it but I thought, you did it purposely to solve the Default loading issue.  working fine now

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts (hunk 2 above)
Comment (FarajiA) [reply src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts:315]: Gotcha.

**Pull request merged**